client/analytics/report/orders/store/index.js

```diff
@@ -0,0 +1,64 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { registerStore, dispatch } from '@wordpress/data';
+import apiRequest from '@wordpress/api-request';
+
+const DEFAULT_STATE = {
+	orders: [],
+};
+
+registerStore( 'woo-dash', {
+	reducer( state = DEFAULT_STATE, action ) {
+		switch ( action.type ) {
+			case 'SET_ORDERS':
+				return {
+					...state,
+					orders: action.orders,
+				};
+			case 'UPDATE_ORDER':
+				const orderIndex = state.orders.findIndex( order => order.id === action.order.id );
+				const orders = state.orders.map( o => o );
+				orders[ orderIndex ] = action.order;
+				return {
+					...state,
+					orders,
+				};
+		}
+
+		return state;
+	},
+
+	actions: {
+		setOrders( orders ) {
+			return {
+				type: 'SET_ORDERS',
+				orders,
+			};
+		},
+		updateOrder( order ) {
+			return {
+				type: 'UPDATE_ORDER',
+				order,
+			};
+		},
+	},
+
+	selectors: {
+		getOrders( state ) {
+			return state.orders;
+		},
+	},
+
+	resolvers: {
+		async getOrders() {
+			try {
+				const orders = await apiRequest( { path: '/wc/v2/orders' } );
```

Comment:
@aduth is this a good place for the `apiRequest` middleware? send `/orders` or some version-less path then have the middleware add the path?

with Calypso's data layer we'd simply intercept the `WPCOM_HTTP_REQUEST` or `RAW_HTTP_REQUEST` actions with custom Redux middleware and change the path property before passing along the action…

Reply:
>@aduth is this a good place for the apiRequest middleware? send /orders or some version-less path then have the middleware add the path?

Since `apiRequest` is a singleton working in a global space, adding a middleware which prefixes the WooCommerce-specific path prefixes could be problematic.

Maybe `apiRequest` here could point to some internal module which just proxies to the underlying `@wordpress/api-request` with the prefix attached?